src/main/scala/com/microsoft/azure/cosmosdb/spark/schema/CosmosDBRowConverter.scala

```diff
@@ -144,17 +144,15 @@ object CosmosDBRowConverter extends RowConverter[Document]
   def rowToJSONObject(row: Row): JSONObject = {
     val jsonObject: JSONObject = new JSONObject()
     row.schema.fields.zipWithIndex.foreach({
-      case (field, i) if row.isNullAt(i) => if (field.dataType == NullType) jsonObject.remove(field.name)
```

Comment:
Shouldn't this behavior change be considered breaking? Should we make it opt-in via config setting?

Unfortunately Json has null value handling as soft contract - so changing the behavior might surprise customers who "expect" null values to be silently removed

Reply:
I see why are you are saying that it can be considered a change in the behaviour. However CosmosDB format is json and null is a valid json type, so if we were dropping null, it is a bug which was causing this behaviour rather than a contract.

Is there any changelog we can track the changes?
how about adding a line in the changelog saying nulls are preserved now?

Reply:
I would agree that round-tripping should be considered the right behavior (representing nulls as is)

Changelog is here - https://github.com/Azure/azure-cosmosdb-spark/blob/2.4/CHANGELOG.md

Reply:
Added changelog entry.

Reply:
added config option.